lib/agent.js

```diff
@@ -111,6 +117,8 @@ Agent.prototype._doClose = function() {
   for (var k in this._msgIdToReq)
     this._msgIdToReq[k].sender.reset()
 
+  if (this._keep) return;
```

Comment:
Can you please remove this._keep, and move the check on opts.sock in this function?